### ISSUE TITLE
Why `href` and not `ui-sref`

### ISSUE THREAD

**Issue by psi-4ward** [ncuillery/angular-breadcrumb#54]
What is the reason to use `href` and not `ui-sref` within the templates? Leads to a page-reload for me.

**Comment by psi-4ward** [comment]
Ive updated to

ui-router: 0.2.13
angular: 1.3.8
angular-breadcrumb: 0.3.3

works as expected now without page-reload, thanks

decorating the template is also a easy solution
```javascript
app.config(function($breadcrumbProvider) {
  $breadcrumbProvider.setOptions({
    template:
      '<div class="breadcrumb">' +
        '<a ng-repeat="step in steps" class="btn btn-flat" ui-sref="{{step.name}}" ng-disabled="$last">{{step.ncyBreadcrumbLabel}}</a>' +
      '</div>'
  });
});
```

**Comment by ncuillery** [comment]
I keep the issue open (to keep in mind the refactoring of the template).

Please, can you give me your old versions of dependencies ? (I would like to know if there is a general problem with a specific set of versions)

**Comment by psi-4ward** [comment]
Sorry to say but i dont have the old dep-list. This project was versioned after the update.

**Comment by stumpdk** [comment]
It is still relevant to implement the use of ui-sref instead of href.
A manual change in the template as suggested by psi-4ward is possible, but only if static content is used.
A dynamic content parent as the example below isn't handled correct:

ncyBreadcrumb: {
    parent: function($scope){ return 'show({id: ' + $scope.itemId+ '})'; },
    label: 'Details'
}

Or am I missing something here?

**Comment by aragao-diego** [comment]
This still **NOT** working?!
```javascript
ncyBreadcrumb: {
    parent: function($scope){ return 'show({id: ' + $scope.itemId+ '})'; },
    label: 'Details'
}
```

I have the same need!

**Comment by ncuillery** [comment]
That's because I **DON'T** have time to investigate and fix it. Fortunately, PR are welcome.

**Comment by aragao-diego** [comment]
Oh i see, make sense. Your work is great!
Keepgoing!

**Comment by kiddo13** [comment]
@aragao-diego It's not working because the state.name is already parsed into state name without params. For example, passing "show({id: 3})" will be parsed as "show" as the step.name.

@ncuillery you could add conf.ncyBreadcrumbState = stateRef; inside $$addStateInChain function.
So, inside the template we could use ui-sref="{{ step.ncyBreadcrumbState }}" instead.

**Comment by kiddo13** [comment]
https://github.com/kiddo13/angular-breadcrumb/commit/5c2cabc21985789205aaf74f5caae9ddc07f96b3

**Comment by kiddo13** [comment]
I made a Pull Request for this. Hope you can review it.

https://github.com/kiddo13/angular-breadcrumb/commit/5c2cabc21985789205aaf74f5caae9ddc07f96b3

**Comment by AnirudhaGohokar** [comment]
@ncuillery Is the feature added by @kiddo13 merged with main branch? 
Thank You.

**Comment by ncuillery** [comment]
@AnirudhaGohokar Yes :smile:

The fix has been merged (release pretty soon) and I addressed the template enhancement in #142.
